File: Java/1769_Minimum_Number_of_Operations_to_Move_All_Balls_to_Each_Box.java
class Solution {
    public int[] minOperations(String boxes) {
        int n = boxes.length();
        int[] result = new int[n];
        for(int i = 0; i < n; i++){
            if(boxes.charAt(i) == '1'){
                for(int j = 0; j < n; j++){
                    if(i != j){
                        result[j] += Math.abs(j-i);
                    }
                }
            }
        }
        return result;
        /*
        for(int i = 0; i < n; i++){
            int sum = 0;
            for(int j = 0; j < n; j++){
                if(i==j){
                    continue;
                }
                else{
                    if(boxes.charAt(j) == '1'){
                        sum += Math.abs(j-i);
                    }
                }
            }
            result[i] = sum;
        }
        return result;
        */
    }
}

File: Python3/1769_Minimum_Number_of_Operations_to_Move_All_Balls_to_Each_Box.py
class Solution:
    def minOperations(self, boxes: str) -> List[int]:
        n = len(boxes)
        l = [0] * n
        r = [0] * n
        result = []
        ball = int(boxes[0])
        for i in range(1,n):
            l[i] = ball + l[i-1]
            ball += int(boxes[i])
        
        ball = int(boxes[n-1])
        for i in range(n-2, -1, -1):
            r[i] = ball + r[i+1]
            ball += int(boxes[i])
        
        #print(l)
        #print(r)
        for i in range(0, n):
            result.append(l[i] + r[i])
        
        return result
        """
        n = len(boxes)
        result = []
        for i in range(0, n):
            temp = 0
            for j in range(0, n):
                if boxes[j] == '1':
                    temp += abs(j-i)
            result.append(temp)
        return result
        """
    """
    110
 l  013
    100
    210
    001011
    
    0  0 1123
    0  0 0124
           1+1
            2+2
    3  3 3221
    11 8 5310
        1
       2+1
      2+3
    """
